### PR TITLE
add post-checkout conda update hook

### PR DESCRIPTION
**Pull request by tomkinsc**
add a script for `.git/hooks/` to update the active conda environment to the current `requirements-conda.txt` file upon `git checkout` or `git clone`

### CODE REVIEW

File: contrib/post-checkout [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+# if conda is available
+if cmd_exists "conda"; then
+    # if a conda environment is active
+    if [ ! -z "$CONDA_DEFAULT_ENV" ]; then
+        # update the conda environment to install the requirements specified
+        if [ "$(uname)" == "Darwin" ]; then
+            # some conda packages dont exist on OSX
+            cat requirements-conda.txt | grep -v kraken > /tmp/requirements-conda.txt
```
Comment (yesimon) [inline contrib/post-checkout]: mktemp for this temporary file and rm later?

File: contrib/post-checkout [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        else
+            # for linux, just use requirements-conda as-is
+            cp requirements-conda.txt /tmp/requirements-conda.txt
+        fi
+
+        conda install -y -c bioconda --file /tmp/requirements-conda.txt
+    fi
+fi
```
Comment (yesimon) [inline contrib/post-checkout]: missing newline

File: contrib/post-checkout (hunk 1 above)
Comment (tomkinsc) [inline contrib/post-checkout]: added

File: contrib/post-checkout (hunk 2 above)
Comment (tomkinsc) [inline contrib/post-checkout]: added (apparently GitHub doesn't hide out-of-date comments in review mode?)

**Pull request merged**